core/drivers/regulator/regulator_gpio.c

```diff
@@ -175,13 +202,25 @@ static TEE_Result get_voltage_level_gpio(const void *fdt, int node,
 
 	if (fdt32_to_cpu(*(cuint + 1))) {
 		assert(!fdt32_to_cpu(*(cuint + 3)));
-		regu->voltage_levels_uv[0] = fdt32_to_cpu(*(cuint + 2));
-		regu->voltage_levels_uv[1] = fdt32_to_cpu(*(cuint));
+		level1 = fdt32_to_cpu(*(cuint));
+		level0 = fdt32_to_cpu(*(cuint + 2));
 	} else {
 		assert(fdt32_to_cpu(*(cuint + 3)) == 1);
-		regu->voltage_levels_uv[0] = fdt32_to_cpu(*(cuint));
-		regu->voltage_levels_uv[1] = fdt32_to_cpu(*(cuint + 2));
+		level0 = fdt32_to_cpu(*(cuint));
+		level1 = fdt32_to_cpu(*(cuint + 2));
+	}
+
+	/* Get the 2 supported levels in increasing order */
+	regu->levels_desc.type = VOLTAGE_TYPE_FULL_LIST;
+	regu->levels_desc.num_levels = 2;
+	if (level0 < level1) {
+		regu->voltage_levels_uv[0] = level0;
+		regu->voltage_levels_uv[1] = level1;
 		regu->voltage_level_high = true;
+	} else {
+		regu->voltage_levels_uv[0] = level1;
+		regu->voltage_levels_uv[1] = level0;
+		regu->voltage_level_high = false;
```

Comment:
actually sorry, with is this:

`regu->voltage_level_high = false;`

instead of
`regu->voltage_level_high = true;`

It is sorted, isn't it?

Reply:
`level0` relates to the voltage level selected by GPIO state 0 and `level1` to the one selected by GPIO state 1.

`regu->voltage_levels_uv[0]` relates the to lower supported voltage level while `regu->voltage_levels_uv[1]` relates the to higher supported voltage level. This it because supported voltage level list must be sorted in increasing level value, see [`enum voltage_type`](https://github.com/OP-TEE/optee_os/blob/4.1.0/core/include/drivers/regulator.h#L54-L56).

Hence `regu->voltage_level_high` stores whether GPIO state 1 relate to the higher level or to the lower level value.